AsyncDisplayKit/TextKit/ASTextKitFontSizeAdjuster.mm

```diff
@@ -179,23 +181,42 @@ - (CGFloat)scaleFactor
     
     if (_attributes.maximumNumberOfLines > 0) {
       // get the number of lines in our possibly scaled string
-      NSUInteger numberOfLines = [self lineCountForString:textStorage];
+      NSUInteger numberOfLines = [self lineCountForString:scaledString];
       if (numberOfLines > _attributes.maximumNumberOfLines) {
         
         for (NSUInteger index = scaleIndex; index < scaleFactors.count; index++) {
-          NSMutableAttributedString *entireAttributedString = [[NSMutableAttributedString alloc] initWithAttributedString:textStorage];
-          [[self class] adjustFontSizeForAttributeString:entireAttributedString withScaleFactor:[scaleFactors[index] floatValue]];
-          
+          scaledString = [[NSMutableAttributedString alloc] initWithAttributedString:textStorage];
```

Comment:
I keep scaled string alive outside of the loops so that i can check line count/text height on the string that may have been scaled by the previous check. I added a comment to make this clearer.